### ISSUE TITLE
Particle initialization

### ISSUE THREAD

**Issue by tmiethlinger** [ECP-WarpX/WarpX#3658]
Hello,

I have two questions / possible issues regarding particle initialization, for `WarpX_DIMS=1`.

### 1.) NUniformPerCell

My input file:
[input_NUniformPerCell.txt](https://github.com/ECP-WarpX/WarpX/files/10551749/input_NUniformPerCell.txt)

> NUniformPerCell: injection with a fixed number of evenly-spaced particles per cell

Using NUniformPerCell, I expected evenly-spaced particles per cell, which is however not the case.

If I plot the particle positions, then they are all located at their respective cell center: 
![image](https://user-images.githubusercontent.com/91594368/215903596-aed74c47-5643-4d13-8213-26eed35fd29b.png)
The "step size" in the staircase shape comes from ppc=500

Plot script:
```
sim = 1
simpath = campaign_dir + "/runs/" + str(sim)
filepath = simpath + "/diags/openPMD/openpmd_" + s + ".h5"
f = h5py.File(filepath, 'r')
Rz = f['data/' + str(i) + '/particles/electron/position/z'][()]
```

### 2.) NRandomPerCell
Basically the same input script as above, but with
```
NUniformPerCell -> NRandomPerCell
num_particles_per_cell_each_dim -> num_particles_per_cell
```
[input_NRandomPerCell.txt](https://github.com/ECP-WarpX/WarpX/files/10551807/input_NRandomPerCell.txt)

I don't know if bug or intended, but if I do the same simulation but different `warpx.random_seed` values, I end up with the same initial particle positions.
![image](https://user-images.githubusercontent.com/91594368/215905221-c928a516-a9ab-4b73-a8be-e37dff8bbb0a.png)

My script to read the data, and compare Rz1 with Rz2. Since they have different seeds, I'd have expected different values.
```
i = 0 # Inspect initial particle positions
s = str(i).zfill(6)

sim = 1
simpath = campaign_dir + "/runs/" + str(sim)
filepath = simpath + "/diags/openPMD/openpmd_" + s + ".h5"
f = h5py.File(filepath, 'r')
Rz1 = f['data/' + str(i) + '/particles/electron/position/z'][()]

sim = 2
simpath = campaign_dir + "/runs/" + str(sim)
filepath = simpath + "/diags/openPMD/openpmd_" + s + ".h5"
f = h5py.File(filepath, 'r')
Rz2 = f['data/' + str(i) + '/particles/electron/position/z'][()]
```

If this is intended behavior, then please allow me to remark that this is not very intuitive.

Thanks, and best regards,
Thomas

**Comment by dpgrote** [comment]
For NUniformPerCell, I've just noticed the same thing, that all of the particles in each cell are placed at the center of the cell. This is fixed in PR #3659

**Comment by RemiLehe** [comment]
@tmiethlinger The above PR has now been merged. Feel free to download the latest version of WarpX (e.g. with `git pull`) and recompile :)

**Comment by ax3l** [comment]
We'll close for now as fixed. Please feel free to comment or reopen if you still see the bug.

**Comment by tmiethlinger** [comment]
Hi @ax3l and @RemiLehe,

thanks, yes it (NUniformPerCell) works for me as intended!
Just one question: Is it expected behavior that WarpX (in `DIM1`, didn't try in other dimensions) creates identical particle position configurations when using `NRandomPerCell`, regardless of the seed? Just asking because other PIC codes (PIConGPU, Smilei) handle that differently.

Thanks,
Thomas

**Comment by dpgrote** [comment]
Hi @tmiethlinger - I'm not able to reproduce the issue with NRandomPerCell. If I run your input file with different seeds, I get different values for the positions. Are you running on CPU or GPU? (I only tested on CPU.)

**Comment by tmiethlinger** [comment]
Hi @dpgrote I used GPU, i.e.:

```
# install
module load CUDA/11.8.0
export AMREX_CUDA_ARCH=8.0 #A100
-DWarpX_COMPUTE=CUDA

# run
#SBATCH --gpus-per-task=1
#SBATCH --gpu-bind=single:1
srun -n 1 $HOME/warpx_1d_test/build/bin/warpx.1d.MPI.CUDA.DP.PDP.OPMD.QED $INPUT

# input
warpx.numprocs = 1
```

**Comment by dpgrote** [comment]
The NRandomPerCell issue should be fixed in PR #3682. When the seed was being specified, it was not setting the seed for the GPU, only for the CPU. This affects all places, not just the 1D particle load.
